src/run_macros/infer_route.cr

```diff
@@ -51,32 +52,67 @@ class InferRoute
     end
   end
 
-  private def resource_pieces
-    if ["index", "create"].includes? action_name
+  private def plural_resource_pieces
+    case action_name
+    when "index", "create"
       [resource]
-    elsif action_name == "new"
+    when "new"
       [resource, "new"]
-    elsif action_name == "edit"
+    when "edit"
       [resource, ":id", "edit"]
-    elsif ["show", "update", "delete"].includes? action_name
+    when "show", "update", "delete"
       [resource, ":id"]
     else
+      resource_error
+    end
+  end
+
+  private def singular_resource_pieces
+    case action_name
+    when "index"
       puts(
         <<-ERROR
         Could not infer route for #{action_class_name}
 
-        Got:
-          #{action_class_name} (missing a known resourceful action)
-
-        Expected something like:
-          Users::Index # Index, Show, New, Create, Edit, Update, or Delete
+        Reason: Singular routes do not support an index action"
         ERROR
       )
 
-      raise "Problem inferring route"
+      raise "Invalid index action for singular resource"
```

Comment:
One small thing, could you capitalize `Index`?

Reply:
Sure, makes sense.